tests/plugins/test_pin_operator_digests.py

```diff
@@ -713,26 +713,6 @@ def test_replacement_opt_out(self, pin_digest, replace_repo, replace_registry,
         # plugin must always retun pullspecs
         assert result['pin_operator_digest']['related_images']['pullspecs']
 
-    @pytest.mark.parametrize('has_envs', [True, False])
-    def test_exclude_csvs(self, workflow, source_dir, caplog, has_envs):
-        # Worker does not care if there is a conflict between relatedImages
-        # and RELATED_IMAGE_* env vars, orchestrator should have caught this already
-        manifests_dir = source_dir.joinpath(OPERATOR_MANIFESTS_DIR)
-        manifests_dir.mkdir()
-        csv = mock_operator_csv(manifests_dir, 'csv.yaml', ['foo'],
-                                with_related_images=True,
-                                with_related_image_envs=has_envs)
-        original_content = csv.read_text("utf-8")
-
-        user_config = get_user_config(OPERATOR_MANIFESTS_DIR)
-        runner = mock_env(workflow, source_dir, site_config=get_site_config(),
-                          user_config=user_config)
-        runner.run()
-
-        assert "Replacing pullspecs" not in caplog.text
-        assert "Creating relatedImages section" not in caplog.text
-        assert csv.read_text("utf-8") == original_content
```

Comment:
This test checked that OSBS will skip CSV files if they already have a relatedImages section. Is that still tested somewhere?

Reply:
It's sort of a weird situation with this testcase. It was ensuring that the `run_in_worker` didn't fail if there was a conflict between `relateImages` and env variables of the same since it should have been caught in `run_in_orchestrator`(it throws run time error if skip_all is False and if it's True the worker doesn't run.). This didn't make sense to me since `run_in_orchestrator` was always running before `run_in_worker` and this testcase was simulating something that wasn't actually supposed to happen.

Furthermore, in OSBS2 `run_in_orchestrator` and `run_in_worker` are merged in the `run` method so `run_in_orchestrator` steps would always execute before `run_in_worker` steps which invalidate this testcase.

Reply:
I think the comment at the top is the confusing part here. The test doesn't check the conflict handling, it checks that the worker part of the plugin will not touch CSV files if they already have `relatedImages`, regardless of whether there is a conflict or not. I think the fix for this test would be to remove the `has_envs` parametrization and set it to False (so that there is no conflict). The conflict handling should already have its own test I think.

Reply:
Oh I see. It makes sense now. Updated the testcase instead of removing it.